.github/actions/setup-dependencies/action.yml

```diff
@@ -0,0 +1,41 @@
+inputs:
+  python-version:
+    description: "Python version to setup"
+    required: false
+    default: "3.8"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Restore pip cache
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: pip-${{ inputs.python-version }}
+
+    - name: Pip install
+      run: python -m pip install --upgrade pip poetry
+      shell: bash
+
+    - name: Restore poetry cache
```

Comment:
I don't think build is using any of the cache directories

Reply:
Indeed. Removed cache handling.